src/coreclr/vm/ilmarshalers.cpp

```diff
@@ -3901,14 +3899,16 @@ void ILNativeArrayMarshaler::EmitCreateMngdMarshaler(ILCodeStream* pslILEmit)
     dwFlags |= (((DWORD)mops.bestfitmapping)        << 16);
     dwFlags |= (((DWORD)mops.throwonunmappablechar) << 24);
 
+    pslILEmit->EmitLDC(dwFlags);
+
     if (!IsCLRToNative(m_dwMarshalFlags) && IsOut(m_dwMarshalFlags) && IsIn(m_dwMarshalFlags))
     {
-        // Unmanaged->managed in/out is the only case where we expect the native buffer to contain valid data.
-        _ASSERTE((dwFlags & MngdNativeArrayMarshaler::FLAG_NATIVE_DATA_VALID) == 0);
-        dwFlags |= MngdNativeArrayMarshaler::FLAG_NATIVE_DATA_VALID;
+        pslILEmit->EmitLDC(1);
+    }
+    else
+    {
+        pslILEmit->EmitLDC(0);
```

Comment:
Is this `true` / `false`? Can we get these doc'd so we know for sure.

Reply:
```suggestion
        pslILEmit->EmitLDC(1); // true
    }
    else
    {
        pslILEmit->EmitLDC(0); // false
```